sdk/python/feast/infra/online_stores/snowflake.py

```diff
@@ -0,0 +1,232 @@
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple
+
+import pandas as pd
+import pytz
+from pydantic import Field
+from pydantic.schema import Literal
+
+from feast import Entity, FeatureView
+from feast.infra.key_encoding_utils import serialize_entity_key
+from feast.infra.online_stores.online_store import OnlineStore
+from feast.infra.utils.snowflake_utils import get_snowflake_conn, write_pandas_binary
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.repo_config import FeastConfigBaseModel, RepoConfig
+from feast.usage import log_exceptions_and_usage
+
+
+class SnowflakeOnlineStoreConfig(FeastConfigBaseModel):
+    """ Online store config for Snowflake """
+
+    type: Literal["snowflake.online"] = "snowflake.online"
+    """ Online store type selector"""
+
+    config_path: Optional[str] = (
+        Path(os.environ["HOME"]) / ".snowsql/config"
+    ).__str__()
+    """ Snowflake config path -- absolute path required (Can't use ~)"""
+
+    account: Optional[str] = None
+    """ Snowflake deployment identifier -- drop .snowflakecomputing.com"""
+
+    user: Optional[str] = None
+    """ Snowflake user name """
+
+    password: Optional[str] = None
+    """ Snowflake password """
+
+    role: Optional[str] = None
+    """ Snowflake role name"""
+
+    warehouse: Optional[str] = None
+    """ Snowflake warehouse name """
+
+    database: Optional[str] = None
+    """ Snowflake database name """
+
+    schema_: Optional[str] = Field("PUBLIC", alias="schema")
+    """ Snowflake schema name """
+
+    class Config:
+        allow_population_by_field_name = True
+
+
+class SnowflakeOnlineStore(OnlineStore):
+    @log_exceptions_and_usage(online_store="snowflake")
+    def online_write_batch(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        data: List[
+            Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        assert isinstance(config.online_store, SnowflakeOnlineStoreConfig)
+
+        dfs = [None] * len(data)
+        for i, (entity_key, values, timestamp, created_ts) in enumerate(data):
+
+            df = pd.DataFrame(
+                columns=[
+                    "entity_feature_key",
+                    "entity_key",
+                    "feature_name",
+                    "value",
+                    "event_ts",
+                    "created_ts",
+                ],
+                index=range(0, len(values)),
+            )
+
+            timestamp = _to_naive_utc(timestamp)
+            if created_ts is not None:
+                created_ts = _to_naive_utc(created_ts)
+
+            for j, (feature_name, val) in enumerate(values.items()):
+                df.loc[j, "entity_feature_key"] = serialize_entity_key(
```

Comment:
what is the purpose of this key?

Reply:
Hybrid tables requires the key of the table to be unique. This can be removed for regular tables. Let me know what you want to do here